config_presto.go

```diff
@@ -0,0 +1,128 @@
+package evergreen
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"github.com/prestodb/presto-go-client/presto"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type PrestoConfig struct {
+	BaseURI           string            `bson:"base_uri" json:"base_uri" yaml:"base_uri"`
+	TLS               bool              `bson:"tls" json:"tls" yaml:"tls"`
+	Port              int               `bson:"port" json:"port" yaml:"port"`
+	Username          string            `bson:"username" json:"username" yaml:"username"`
+	Password          string            `bson:"password" json:"password" yaml:"password"`
+	Source            string            `bson:"source" json:"source" yaml:"source"`
+	Catalog           string            `bson:"catalog" json:"catalog" yaml:"catalog"`
+	Schema            string            `bson:"schema" json:"schema" yaml:"schema"`
+	SessionProperties map[string]string `bson:"session_properties" json:"session_properties" yaml:"session_properties"`
+
+	db *sql.DB
+}
+
+var (
+	PrestoConfigBaseURIKey           = bsonutil.MustHaveTag(PrestoConfig{}, "BaseURI")
+	PrestoConfigPortKey              = bsonutil.MustHaveTag(PrestoConfig{}, "Port")
+	PrestoConfigUsernameKey          = bsonutil.MustHaveTag(PrestoConfig{}, "Username")
+	PrestoConfigPasswordKey          = bsonutil.MustHaveTag(PrestoConfig{}, "Password")
+	PrestoConfigSourceKey            = bsonutil.MustHaveTag(PrestoConfig{}, "Source")
+	PrestoConfigCatalogKey           = bsonutil.MustHaveTag(PrestoConfig{}, "Catalog")
+	PrestoConfigSchemaKey            = bsonutil.MustHaveTag(PrestoConfig{}, "Schema")
+	PrestoConfigSessionPropertiesKey = bsonutil.MustHaveTag(PrestoConfig{}, "SessionProperties")
+)
+
+func (*PrestoConfig) SectionId() string { return "presto" }
+
+func (c *PrestoConfig) Get(env Environment) error {
+	ctx, cancel := env.Context()
+	defer cancel()
+	coll := env.DB().Collection(ConfigCollection)
+
+	res := coll.FindOne(ctx, byId(c.SectionId()))
+	if err := res.Err(); err != nil {
+		if err == mongo.ErrNoDocuments {
+			*c = PrestoConfig{}
+			return nil
+		}
+		return errors.Wrapf(err, "retrieving section '%s'", c.SectionId())
+	}
+
+	if err := res.Decode(c); err != nil {
+		return errors.Wrapf(err, "decoding section '%s'", c.SectionId())
+	}
+
+	if err := c.setupDB(ctx); err != nil {
```

Comment:
I don't know if it would really make a difference since this gets called during env setup anyways. One thing I don't want to do here is cause an error on startup, this shouldn't cause the entire application to go down. The alert should suffice so we can quickly revert or make changes. I am not super opinionated though.

Reply:
So I made sure in the last 6 week review that we will make it a clear goal to remove this API in the follow-up project: https://docs.google.com/document/d/1sSXjKenryeVpnieqI_txPnlF9o6ddfVwVCaFClPz_Cc/edit#heading=h.6olbmpn60t0z. So once that project is complete this API endpoint will die and we can remove this code.